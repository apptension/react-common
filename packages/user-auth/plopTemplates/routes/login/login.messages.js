import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'login.title',
    defaultMessage: 'Sign in',
  },
  passwordLabel: {
    id: 'login.fields.password.label',
    defaultMessage: 'Password',
  },
  passwordPresenceError: {
    id: 'login.fields.password.error.presence',
    defaultMessage: 'Password cannot be blank',
  },
  passwordWrongError: {
    id: 'login.fields.password.error.wrong',
    defaultMessage: 'Unable to log in with provided credentials',
  },
  emailLabel: {
    id: 'login.fields.email.label',
    defaultMessage: 'Email',
  },
  emailPresenceError: {
    id: 'login.fields.email.error.presence',
    defaultMessage: 'Email cannot be blank',
  },
  submit: {
    id: 'login.submit',
    defaultMessage: 'Sign in',
  },
  recoverPassword: {
    id: 'login.recoverPassword',
    defaultMessage: 'Forgotten your password?',
  },
  registerPrompt: {
    id: 'login.registerPrompt',
    defaultMessage: 'I do not have an account.',
  },
  signUp: {
    id: 'login.signUp',
    defaultMessage: 'Register',
  },
});
