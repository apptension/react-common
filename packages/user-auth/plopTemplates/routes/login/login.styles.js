import styled from 'styled-components';
import { prop } from 'ramda';

export const Container = styled.div`
  width: 100%;
`;

export const TextField = styled.input`
  width: 100%;
`;

export const FieldLine = styled.div`
  display: flex;
`;

export const Form = styled.form`
  width: 100%;
`;

export const ButtonGroup = styled.div`
  display: flex;
  justify-content: space-between;
`;

export const RegisterBar = styled.div`
  text-align: left;
  width: 100%;
`;
