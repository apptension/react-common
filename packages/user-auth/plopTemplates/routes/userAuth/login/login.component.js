import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

import {
  Container,
  Header,
  Form,
  ButtonGroup,
} from '../../../components/forms/form.styles';
import messages from './login.messages';

import { TextField } from '../../../components/forms/textField';

export class Login extends PureComponent {
  static propTypes = {
    language: PropTypes.string,
    intl: PropTypes.object.isRequired,
    login: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired,
  };

  handleSubmit = (values) => new Promise(() => {
    this.props.login(
      values.get('email'),
      values.get('password')
    );
  });

  render() {
    const { intl, handleSubmit, language } = this.props;

    return (
      <Container>
        <Header>
          <FormattedMessage {...messages.title} />
        </Header>

        <Form onSubmit={handleSubmit(this.handleSubmit)} noValidate>
          <Field
            component={TextField}
            name="email"
            type="email"
            label={messages.emailLabel}
            messages={messages}
            intl={intl}
          />

          <Field
            component={TextField}
            name="password"
            type="password"
            label={messages.passwordLabel}
            messages={messages}
            intl={intl}
          />

          <ButtonGroup>
            <button type="submit">
              <FormattedMessage {...messages.submit} />
            </button>
            <Link to={`/${language}/recover-password`}>
              <FormattedMessage {...messages.recoverPassword} />
            </Link>
            <Link to={`/${language}/register`}>
              <FormattedMessage {...messages.signUp} />
            </Link>
          </ButtonGroup>
        </Form>
      </Container>
    );
  }
}
