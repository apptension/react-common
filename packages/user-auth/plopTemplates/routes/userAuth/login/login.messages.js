import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'login.title',
    defaultMessage: 'Sign in',
  },
  passwordLabel: {
    id: 'login.fields.password.label',
    defaultMessage: 'Password',
  },
  passwordPresenceError: {
    id: 'login.fields.password.error.presence',
    defaultMessage: 'Password cannot be blank',
  },
  passwordWrongError: {
    id: 'login.fields.password.error.wrong',
    defaultMessage: 'Unable to log in with provided credentials',
  },
  emailLabel: {
    id: 'login.fields.email.label',
    defaultMessage: 'Email',
  },
  emailPresenceError: {
    id: 'login.fields.email.error.presence',
    defaultMessage: 'Email cannot be blank',
  },
  emailFormatError: {
    id: 'login.fields.email.error.email',
    defaultMessage: 'Email is invalid',
  },
  submit: {
    id: 'login.submit',
    defaultMessage: 'Sign in',
  },
  recoverPassword: {
    id: 'login.recoverPassword',
    defaultMessage: 'Forgotten your password?',
  },
  signUp: {
    id: 'login.signUp',
    defaultMessage: 'Register',
  },
});
