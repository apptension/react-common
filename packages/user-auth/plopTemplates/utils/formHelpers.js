import reportError from 'report-error';
import { List } from "immutable";
import { when, is, pipe } from 'ramda';

export const getFormFieldHelperText = (intl, messages, { error, invalid, touched }) => {
  if (invalid && touched) {
    try {
      return intl.formatMessage(messages[pipe(
        when(
          is(List),
          (errors) => errors.first()
        ),
        when(
          is(Array),
          ([firstError]) => firstError
        ),
      )(error)]);
    } catch (e) {
      reportError(e);
    }
  }
  return '';
};
