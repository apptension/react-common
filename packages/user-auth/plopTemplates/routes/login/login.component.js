import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form/immutable';
import { FormattedMessage } from 'react-intl';
import { Link } from 'react-router-dom';

import { getFormFieldHelperText } from '../../utils/formHelpers';
import {
  Container,
  TextField,
  Form,
  ButtonGroup,
  RegisterBar,
  FieldLine,
} from './login.styles';
import messages from './login.messages';

export class Login extends PureComponent {
  static propTypes = {
    intl: PropTypes.object.isRequired,
    login: PropTypes.func.isRequired,
    handleSubmit: PropTypes.func.isRequired,
  };

  handleSubmit = (values) => new Promise(() => {
    this.props.login(
      values.get('email'),
      values.get('password')
    );
  });

  renderTextField = ({ meta, input, type, label }) => {
    return (
      <FieldLine>
        <TextField
          error={meta.invalid && meta.touched}
          value={input.value}
          type={type}
          label={label}
          helperText={getFormFieldHelperText(this.props.intl, messages, meta)}
          InputProps={input}
        />
      </FieldLine>
    );
  };

  render() {
    const { intl, handleSubmit } = this.props;

    return (
      <Container>
        <FormattedMessage {...messages.title} />

        <Form onSubmit={handleSubmit(this.handleSubmit)} noValidate>
          <Field
            component={this.renderTextField}
            name="email"
            type="email"
            label={intl.formatMessage(messages.emailLabel)}
          />

          <Field
            component={this.renderTextField}
            name="password"
            type="password"
            label={intl.formatMessage(messages.passwordLabel)}
          />

          <ButtonGroup>
            <Link to="/recover-password">
              <FormattedMessage {...messages.recoverPassword} />
            </Link>
            <button type="submit">
              <FormattedMessage {...messages.submit} />
            </button>
          </ButtonGroup>
          <RegisterBar>
            <FormattedMessage {...messages.registerPrompt} />
            <Link to="/register">
              <FormattedMessage {...messages.signUp} />
            </Link>
          </RegisterBar>
        </Form>
      </Container>
    );
  }
}
