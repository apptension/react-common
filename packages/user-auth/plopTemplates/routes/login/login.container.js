import { reduxForm } from 'redux-form/immutable';
import validate from 'validate.js';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { injectIntl } from 'react-intl';

import { UserAuthActions, LOGIN_FORM } from '../../modules/userAuth';
import { Login } from './login.component';

const mapStateToProps = createStructuredSelector({});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  login: UserAuthActions.login,
}, dispatch);

export default compose(
  injectIntl,
  connect(mapStateToProps, mapDispatchToProps),
  reduxForm({
    form: LOGIN_FORM,
    validate: (values) => validate(values.toJS(), {
      email: {
        presence: {
          message: '^emailPresenceError',
        },
      },
      password: {
        presence: {
          message: '^passwordPresenceError',
        },
      },
    }),
  }),
)(Login);
