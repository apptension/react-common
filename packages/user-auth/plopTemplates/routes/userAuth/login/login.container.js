import { reduxForm } from 'redux-form/immutable';
import validate from 'validate.js';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'ramda';
import { injectIntl } from 'react-intl';

import { UserAuthActions, LOGIN_FORM } from '../../../modules/userAuth';
import { Login } from './login.component';
import { selectLocalesLanguage } from '../../../modules/locales/locales.selectors';

const mapStateToProps = createStructuredSelector({
  language: selectLocalesLanguage,
});

export const mapDispatchToProps = (dispatch) => bindActionCreators({
  login: UserAuthActions.login,
}, dispatch);

export default compose(
  injectIntl,
  connect(mapStateToProps, mapDispatchToProps),
  reduxForm({
    form: LOGIN_FORM,
    validate: (values) => validate(values.toJS(), {
      email: {
        presence: {
          message: '^emailPresenceError',
        },
        email: {
          message: '^emailFormatError',
        },
      },
      password: {
        presence: {
          message: '^passwordPresenceError',
        },
      },
    }),
  }),
)(Login);
