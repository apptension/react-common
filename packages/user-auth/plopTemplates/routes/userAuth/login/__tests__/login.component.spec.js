import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';
import { expect } from 'chai';
import { Map } from 'immutable';
import { identity } from 'ramda';

import { intlMock } from '../../../../utils/testUtils';
import { Login } from '../login.component';
import { Form } from '../../../../components/forms/form.styles';

describe('<Login />', () => {
  const defaultProps = {
    intl: intlMock(),
    handleSubmit: identity,
    login: identity,
  };

  const component = (props) => shallow(
    <Login {...defaultProps} {...props} />
  );

  it('should render itself', () => {
    const wrapper = component();
    global.expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should call register prop when handling submit', () => {
    const email = 'email@test.com';
    const password = 'passw0rd';

    const login = spy();
    const handleSubmit = spy();
    const wrapper = component({ login, handleSubmit });
    wrapper.find(Form).simulate('submit');

    const [fn] = handleSubmit.getCall(0).args;

    fn(Map({ email, password }));

    expect(login).to.have.been.calledWith(email, password);
  });
});
