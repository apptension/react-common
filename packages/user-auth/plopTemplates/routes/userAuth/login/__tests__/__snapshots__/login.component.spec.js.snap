// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render itself 1`] = `
.c0 {
  width: 100%;
  max-width: 600px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: auto;
}

<div
  className="c0"
>
  <styled.h1>
    <FormattedMessage
      defaultMessage="Sign in"
      id="login.title"
      values={Object {}}
    />
  </styled.h1>
  <styled.form
    noValidate={true}
    onSubmit={[Function]}
  >
    <Field
      component={[Function]}
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      label={
        Object {
          "defaultMessage": "Email",
          "id": "login.fields.email.label",
        }
      }
      messages={
        Object {
          "emailFormatError": Object {
            "defaultMessage": "Email is invalid",
            "id": "login.fields.email.error.email",
          },
          "emailLabel": Object {
            "defaultMessage": "Email",
            "id": "login.fields.email.label",
          },
          "emailPresenceError": Object {
            "defaultMessage": "Email cannot be blank",
            "id": "login.fields.email.error.presence",
          },
          "passwordLabel": Object {
            "defaultMessage": "Password",
            "id": "login.fields.password.label",
          },
          "passwordPresenceError": Object {
            "defaultMessage": "Password cannot be blank",
            "id": "login.fields.password.error.presence",
          },
          "passwordWrongError": Object {
            "defaultMessage": "Unable to log in with provided credentials",
            "id": "login.fields.password.error.wrong",
          },
          "recoverPassword": Object {
            "defaultMessage": "Forgotten your password?",
            "id": "login.recoverPassword",
          },
          "signUp": Object {
            "defaultMessage": "Register",
            "id": "login.signUp",
          },
          "submit": Object {
            "defaultMessage": "Sign in",
            "id": "login.submit",
          },
          "title": Object {
            "defaultMessage": "Sign in",
            "id": "login.title",
          },
        }
      }
      name="email"
      type="email"
    />
    <Field
      component={[Function]}
      intl={
        Object {
          "formatMessage": [Function],
        }
      }
      label={
        Object {
          "defaultMessage": "Password",
          "id": "login.fields.password.label",
        }
      }
      messages={
        Object {
          "emailFormatError": Object {
            "defaultMessage": "Email is invalid",
            "id": "login.fields.email.error.email",
          },
          "emailLabel": Object {
            "defaultMessage": "Email",
            "id": "login.fields.email.label",
          },
          "emailPresenceError": Object {
            "defaultMessage": "Email cannot be blank",
            "id": "login.fields.email.error.presence",
          },
          "passwordLabel": Object {
            "defaultMessage": "Password",
            "id": "login.fields.password.label",
          },
          "passwordPresenceError": Object {
            "defaultMessage": "Password cannot be blank",
            "id": "login.fields.password.error.presence",
          },
          "passwordWrongError": Object {
            "defaultMessage": "Unable to log in with provided credentials",
            "id": "login.fields.password.error.wrong",
          },
          "recoverPassword": Object {
            "defaultMessage": "Forgotten your password?",
            "id": "login.recoverPassword",
          },
          "signUp": Object {
            "defaultMessage": "Register",
            "id": "login.signUp",
          },
          "submit": Object {
            "defaultMessage": "Sign in",
            "id": "login.submit",
          },
          "title": Object {
            "defaultMessage": "Sign in",
            "id": "login.title",
          },
        }
      }
      name="password"
      type="password"
    />
    <styled.div>
      <button
        type="submit"
      >
        <FormattedMessage
          defaultMessage="Sign in"
          id="login.submit"
          values={Object {}}
        />
      </button>
      <Link
        replace={false}
        to="/undefined/recover-password"
      >
        <FormattedMessage
          defaultMessage="Forgotten your password?"
          id="login.recoverPassword"
          values={Object {}}
        />
      </Link>
      <Link
        replace={false}
        to="/undefined/register"
      >
        <FormattedMessage
          defaultMessage="Register"
          id="login.signUp"
          values={Object {}}
        />
      </Link>
    </styled.div>
  </styled.form>
</div>
`;
